src/main/java/sg/ncl/AppConfig.java

```diff
@@ -1,21 +1,28 @@
 package sg.ncl;
 
 import nz.net.ultraq.thymeleaf.LayoutDialect;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.autoconfigure.web.ErrorAttributes;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
+import javax.inject.Inject;
+
 /**
  * @author by Te Ye
  */
 @Configuration("sg.ncl.AppConfig")
 public class AppConfig {
-    @Autowired
+
+    @Inject
     private ErrorAttributes errorAttributes;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use constructor injection for this field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3306)